apps/api/src/controllers/post.ts

```diff
@@ -0,0 +1,120 @@
+import Post from "../models/post";
+import Comment from "../models/comment";
+
+const getPosts = async (req, res, next) => {
+  try {
+    const posts = await Post.find();
+    res.status(200).json(posts);
+  } catch (e) {
+    next(e);
+  }
+};
+
+const getPostsByCategory = async (req, res, next) => {
+  try {
+    const { category } = req.params;
+    const posts = await Post.find();
```

Comment:
You can do the find with where clause to avoid doing the filter manually.
```
Post.where('category').equals(categoryId).populate('category');
```
or
```
Post.find({ category }).populate('category')
```